file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,523 @@
+      " /AIF/IS_HIDDEN, swap to positive stature
+      "! <p class="shorttext">Show Tree Node</p>
+      "! Show tree node in view 1 tree
+      "! $default 'X'
+      is_tree_node_visible        TYPE abap_bool,
+      " (type /aif/alertrelevant)
+      "! <p class="shorttext">Relevant for Alert Recipient Determination</p>
+      "! Relevant for alert recipient determination
+      is_alert_recipient_relevant TYPE abap_bool,
+    END OF ty_key_field_selection.
+
+  TYPES:
+    "! <p class="shorttext">SAP Fiori Features</p>
+    "! SAP Fiori specific features
+    BEGIN OF ty_fiori_features,
+      " (type /aif/semantic_object)
+      "! <p class="shorttext">Semantic Object</p>
+      "! Semantic object
+      semantic_object             TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/semantic_action)
+      "! <p class="shorttext">Semantic Action</p>
+      "! Semantic action
+      semantic_action             TYPE zif_aff_types_v1=>ty_description_60,
+      " /AIF/NOT_SHOW_FIELD
+      "! <p class="shorttext">Display in Message Monitoring</p>
+      "! Display in message monitoring
+      "! $default 'X'
+      is_shown_in_message_monitor TYPE abap_bool,
+      " /AIF/IS_DEFAULT_FIELD
+      "! <p class="shorttext">Display in Message Monitoring by Default</p>
+      "! Display in message monitoring by default
+      is_shown_by_default         TYPE abap_bool,
+    END OF ty_fiori_features.
+
+  TYPES:
+    " Related key fields used by key field rules
+    "! <p class="shorttext">Rule Key Field</p>
+    "! Rule key field
+    BEGIN OF ty_rule_key_field,
+      " (type /aif/key_fieldname_enh)
+      "! <p class="shorttext">Rule Key Field Name</p>
+      "! Rule Key field name
+      rule_key_field_name  TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/field_number_enh)
+      "! <p class="shorttext">Field Sequence Number</p>
+      "! Field sequence number
+      field_sequnce_number TYPE c LENGTH 3,
+    END OF ty_rule_key_field.
+
+  "! <p class="shorttext">Rule Key Fields</p>
+  "! Rule key fields
+  TYPES ty_rule_key_fields TYPE STANDARD TABLE OF ty_rule_key_field WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">Key Field Rule</p>
+    "! Key field rule
+    BEGIN OF ty_key_field_rule,
+      "! <p class="shorttext">Rule Key Fields</p>
+      "! Rule key fields
+      rule_key_fields TYPE ty_rule_key_fields,
+      "! <p class="shorttext">Key Field Rule Class</p>
+      "! Key Field Rule class
+      rule_class      TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Key Field Rule Method</p>
+      "! Key field rule method
+      rule_method     TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_key_field_rule.
+
+  TYPES:
+    "! <p class="shorttext">Key Field</p>
+    "! Key field
+    BEGIN OF ty_key_field,
+      "! <p class="shorttext">General</p>
+      "! General key field settings
+      key_field_settings  TYPE ty_key_field_settings,
```

Comment:
What about
```suggestion
      "! <p class="shorttext">Key Field Details</p>
      "! General key field settings
      ky_field_details  TYPE ty_key_field_settings,
```
? This would match to our usual recommendation with Plural (Key Fields) for Tables and Singular + Details for the Substructure (Key Field Details)